src/components/Table/sortRows.ts

```diff
@@ -9,37 +9,35 @@ import {
 } from "src/components/Table/GridTable";
 import { SortState } from "src/components/Table/useSortState";
 
-// We currently mutate `rows` while sorting; this would be bad if rows was directly
-// read from an immutable store like the apollo cache, but we basically always make
-// a copy in the process of adding our `kind` tags.
-//
-// I suppose that is an interesting idea, would we ever want to render a GQL query/cache
-// result directly into the table without first doing a kind-mapping? Like maybe we could
-// use __typename as the kind.
+// Returns a shallow copy of the `rows` parameter sorted based on `sortState`
 export function sortRows<R extends Kinded>(
   columns: GridColumn<R>[],
   rows: GridDataRow<R>[],
   sortState: SortState<number>,
-): void {
-  sortBatch(columns, rows, sortState);
+): GridDataRow<R>[] {
+  const sorted = sortBatch(columns, rows, sortState);
   // Recursively sort child rows
-  for (const row of rows) {
+  for (const row of sorted) {
     if (row.children) {
-      sortRows(columns, row.children, sortState);
+      // Replace the existing `children` prop with a sorted shallow copy
+      Object.assign(row, { children: sortRows(columns, row.children, sortState) });
```

Comment:
Oh damn...we are actually mutating `row` so we're losing original order here...hrm...

Reply:
I think this probably needs to be:

```
sorted.forEach((row, i) => {
  if (row.children) {
    sorted[i] = { ...sorted[i], children: sortRows(...) };
  }
});
```

That way we'll leave the original `row` that is in the original `rows` untouched.

Reply:
Yea, great point. I'm kind of surprised the tests didn't catch this... 🤔

Reply:
Ahh.. I didn't write a test for reverting sort on a nested structure. Will add before merging.